common/src/lib.rs

```diff
@@ -12,6 +12,64 @@ use tempfile::NamedTempFile;
 mod region;
 pub use region::{Block, RegionDefinition, RegionOptions};
 
+#[derive(thiserror::Error, Debug, PartialEq, Clone, Serialize, Deserialize)]
+pub enum CrucibleError {
+    #[error("Error: {0}")]
+    GenericError(String),
+
+    #[error("IO Error: {0}")]
+    IoError(String),
+
+    #[error("data store disconnected")]
+    Disconnect,
+
+    #[error("Error grabbing data lock")]
+    DataLockError,
+
+    #[error("Error grabbing reader-writer {0} lock")]
+    RwLockError(String),
+
+    #[error("BlockReqWaiter recv channel disconnected")]
+    RecvChannelDisconnected,
+
+    #[error("Offset not block aligned")]
+    OffsetNotBlockAligned,
+
+    #[error("Data length not block size multiple")]
+    DataLenNotBlockSizeMultiple,
+
+    #[error("Block size mismatch")]
+    BlockSizeMismatch,
+
+    #[error("Invalid number of blocks: {0}")]
+    InvalidNumberOfBlocks(String),
+
+    #[error("Offset past end of extent")]
+    PastEndOfExtent,
```

Comment:
Nit,
to be consistent, this should be named: OffsetPastEndOfExtent, but that is too long, so in keeping
with my other nits, how about OffsetBad or OffsetInvalid?